Tools/CISupport/ews-build/steps.py

```diff
@@ -3229,6 +3228,157 @@ def send_email_for_pre_existing_failure(self, test_name):
+                                                KillOldProcesses(),
+                                                RunTest262TestsWithoutChange(),
+                                                AnalyzeTest262TestsResults()])
+        return rc
+
+
+class RunTest262TestsWithoutChange(RunTest262Tests):
+    name = 'test262-test-without-change'
+    prefix = 'test262_clean_tree_'
+
+    def evaluateCommand(self, cmd):
+        rc = shell.Test.evaluateCommand(self, cmd)
+        self.setProperty('clean_tree_run_status', rc)
+        return rc
+
+
+class AnalyzeTest262TestsResults(buildstep.BuildStep, AddToLogMixin):
+    name = 'analyze-test262-tests-results'
+    description = ['analyze-test262-test-results']
+    descriptionDone = ['analyze-test262-tests-results']
+    NUM_FAILURES_TO_DISPLAY = 10
+
+    def start(self):
+        stress_failures_with_change = set(self.getProperty('jsc_stress_test_failures', []))
+        binary_failures_with_change = set(self.getProperty('jsc_binary_failures', []))
+        clean_tree_stress_failures = set(self.getProperty('jsc_clean_tree_stress_test_failures', []))
+        clean_tree_binary_failures = set(self.getProperty('jsc_clean_tree_binary_failures', []))
+        clean_tree_failures = list(clean_tree_binary_failures) + list(clean_tree_stress_failures)
+        clean_tree_failures_string = ', '.join(clean_tree_failures[:self.NUM_FAILURES_TO_DISPLAY])
+
+        flaky_stress_failures_with_change = set(self.getProperty('jsc_flaky_and_passed', {}).keys())
+        clean_tree_flaky_stress_failures = set(self.getProperty('jsc_clean_tree_flaky_and_passed', {}).keys())
+        flaky_stress_failures = sorted(list(flaky_stress_failures_with_change) + list(clean_tree_flaky_stress_failures))[:self.NUM_FAILURES_TO_DISPLAY]
+        flaky_failures_string = ', '.join(flaky_stress_failures)
+
+        new_stress_failures = stress_failures_with_change - clean_tree_stress_failures
+        new_binary_failures = binary_failures_with_change - clean_tree_binary_failures
+        self.new_stress_failures_to_display = ', '.join(sorted(list(new_stress_failures))[:self.NUM_FAILURES_TO_DISPLAY])
+        self.new_binary_failures_to_display = ', '.join(sorted(list(new_binary_failures))[:self.NUM_FAILURES_TO_DISPLAY])
+
+        self._addToLog('stderr', '\nFailures with change: {}'.format(list(binary_failures_with_change) + list(stress_failures_with_change))[:self.NUM_FAILURES_TO_DISPLAY])
+        self._addToLog('stderr', '\nFlaky Tests with change: {}'.format(', '.join(flaky_stress_failures_with_change)))
+        self._addToLog('stderr', '\nFailures on clean tree: {}'.format(clean_tree_failures_string))
+        self._addToLog('stderr', '\nFlaky Tests on clean tree: {}'.format(', '.join(clean_tree_flaky_stress_failures)))
+
+        if (not stress_failures_with_change) and (not binary_failures_with_change):
+            # If we've made it here, then jsc-tests and re-run-jsc-tests failed, which means
```

Comment:
This comment doesn't seem correct for this queue.

Reply:
I completely agree. But I wasn't certain how to specifically deal with removing all of the JSC Stress test failure lines. Basically everything under `class AnalyzeTest262TestsResults(buildstep.BuildStep, AddToLogMixin):` needs a lot of work, and I'm not entirely certain how to proceed.